#### Homework 9
<br> **1.** Consider:
$$A = \begin{bmatrix}
    1 & -3 & 3\\
    2 & -2 & 2\\
    2 & 0 & 0\\
\end{bmatrix}$$
<br>**1a.** 
<br>(See Attached)
<br>**1b.**

In [1]:
clc, clear, close

%a See Attached

%b
A = [1 -3 3; 2 -2 2; 2 0 0]
[eig_vec_tru,eig_val_tru] = eig(A)

A =
     1    -3     3
     2    -2     2
     2     0     0
eig_vec_tru =
        -0.333333333333333      2.81970303603256e-16        -0.707106781186548
        -0.666666666666667         0.707106781186547     -6.33872908616881e-16
        -0.666666666666667         0.707106781186547         0.707106781186547
eig_val_tru =
                         1                         0                         0
                         0      1.11022302462516e-15                         0
                         0                         0                        -2


**Comments:** The eigenvector associated with $\lambda_1 = 1,\ \mathbf{u_1} = [1, 2, 2]^{T}$ is of opposite sign to the one <br>computed by hand. However, this is equivalent if $\mathbf{A}\mathbf{u_1} = \lambda_1\mathbf{u_1}$. Everything else agrees within $10^{-16}$ precision.

**1c.**
<br>(See Attached)

**2a.**
Save as `power_method.m` in directory.
<br>
```
function [eval, evec] = power_method(A,x,tol)

[n,m] = size(x);
    if length(x) > n
        x = x';
    end

lambB = 0;
lambA = inf;

    while norm(lambB - lambA,inf) >= tol
        x_new = A*x;
        lambA = lambB; lambB = x_new(1)/x(1);
        x = x_new/norm(x,inf);
    end

evec = x/norm(x,2); eval = lambB;
end
```
<br>**2b.**

In [2]:
A = [ 1 -3 3; 2 -2 2; 2 0 0 ];
tol = 10^-3


[evectru,evaltru] = eig(A)

x = [1 1 1]
[eig_val, eig_vec] = power_method(A,x,tol)

x = [1 -1 1]
[eig_val, eig_vec] = power_method(A,x,tol)

tol =
                     0.001
evectru =
        -0.333333333333333      2.81970303603256e-16        -0.707106781186548
        -0.666666666666667         0.707106781186547     -6.33872908616881e-16
        -0.666666666666667         0.707106781186547         0.707106781186547
evaltru =
                         1                         0                         0
                         0      1.11022302462516e-15                         0
                         0                         0                        -2
x =
     1     1     1
eig_val =
     1
eig_vec =
         0.333333333333333
         0.666666666666667
         0.666666666666667
x =
     1    -1     1
eig_val =
         -2.00027468335114
eig_vec =
         0.707155332318325
       6.4739040783507e-05
         -0.70705822375715


**2c.**
<br> (See Attached)

**2d.** Consider:
$$B = \begin{bmatrix}
    -3 & -2\\
    4 & 3\\
\end{bmatrix}$$

In [3]:
B = [-2 -1; 4 3];
x = [1 1];


fprintf('***Computed Using power_method()***\n')
[eig_val, eig_vec] = power_method(B,x,tol)

fprintf('\n\n***Multiplying B*u and Eigval*u ***\n')
matrix_tmes_vec = B*eig_vec, lambda_times_vec = eig_vec*eig_val

fprintf('\n\n***Are they approximately equal within tol?***\n')
abs(B*eig_vec-eig_vec*eig_val)<= tol

fprintf('\n\n***Using eig() to compute Matlab "true" eigen vecs and vals***\n')
[eig_vec_tru,eig_val_tru] = eig(B)

***Computed Using power_method()***
eig_val =
          1.99977113562503
eig_vec =
        -0.242529094134135
         0.970144132847525


***Multiplying B*u and Eigval*u ***
matrix_tmes_vec =
        -0.485085944579256
          1.94031602200604
lambda_times_vec =
        -0.485002681998728
          1.94006623426445


***Are they approximately equal within tol?***
ans =
  2x1 logical array
   1
   1


***Using eig() to compute Matlab "true" eigen vecs and vals***
eig_vec_tru =
        -0.707106781186547         0.242535625036333
         0.707106781186547        -0.970142500145332
eig_val_tru =
    -1     0
     0     2


**Comments:** Again the sign is reversed on $\mathbf{u_2} = [0.242535625036333, -0.970142500145332]^{T}$ when <br>computed using `power_method( )`. However with $\ \lambda_2 = 2\ $ then $\ \mathbf{B}\mathbf{u_2} = \lambda_2\mathbf{u_2}$. 
<br>When comparing to `eig(B)` this roughly confirms that `power_method( )` with initial guess $ \mathbf{x} = [1, 1]^{T}$ does seem to converge 
<br>toward an eigenvector of $B$.